### PR TITLE
 [Enhancement] (nereids)implement showTableCreationCommand in nereids #42776 

### LINKED ISSUE #42776: [Enhancement] (nereids)implement showTableCreationCommand in nereids
showTableCreation command is supported in legacy planner but not in nereids(new doris fe planner). you need implement the following showTableCreation command in nereids:
unsupportedshowTableCreationStatement
``` SHOW TABLE CREATION ((FROM | IN) database=multipartIdentifier)? wildWhere?    ```

### PR DESCRIPTION
**Pull request by rijeshkp**
### What problem does this PR solve?

Issue Number: close #42776

Related PR: #xxx

Problem Summary:
Support for Show table creation command in Nereids

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [ ] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by rijeshkp** [comment]
run buildall

**Comment by rijeshkp** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowTableCreationCommand.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        for (List<Comparable> row : rowSet) {
+            List<String> resultRow = Lists.newArrayList();
+            for (Comparable column : row) {
+                resultRow.add(column.toString());
+            }
+            resultRowSet.add(resultRow);
+            keyNameSet.add(resultRow.get(0));
+        }
+
+        ShowResultSetMetaData showMetaData = getMetaData();
```
Comment (Vallishp) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowTableCreationCommand.java:101]: 1) can directly use META_DATA and no need for getMetaData function.
2) there are few empty lines in code, pls remove it

**Comment by rijeshkp** [comment]
run buildall

**Review by starocean999 — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java [hunk 2]
```diff
@@ -4936,4 +4938,15 @@ public LogicalPlan visitShowCollation(ShowCollationContext ctx) {
...
+
+    @Override
+    public LogicalPlan visitShowTableCreation(ShowTableCreationContext ctx) {
+        String dbName = null;
+        if (ctx.database != null) {
+            List<String> nameParts = visitMultipartIdentifier(ctx.database);
+            dbName = nameParts.get(0); // only one entry possible
+        }
+
+        return new ShowTableCreationCommand(dbName, ctx.STRING_LITERAL().getText());
```
Comment (LiBinfeng-01) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java:4950]: ctx.STRING_LITERAL() is not always true defined by DorisParser.g4. Pls check this behavior and add some cases

Comment (rijeshkp) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java]: Addressed. Please review. Also this command was supported earlier (for iceberg). Looks like the support has been removed later. Now the command just returns ok,.

**Comment by rijeshkp** [comment]
run buildall

**Comment by rijeshkp** [comment]
run buildall

**Review by Vallishp — APPROVED** [review]
LGTM

**Comment by rijeshkp** [comment]
run buildall

**Comment by rijeshkp** [comment]
run buildall

**Comment by rijeshkp** [comment]
run compile

**Comment by rijeshkp** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Pull request merged**